### PR TITLE
Fix eclipselsp path documentation.

### LINKED ISSUE #2468: eclipselsp not working
## Information

**VIM version**
                                                                                                                                                         
NVIM v0.3.5
Build type: Release

Operating System:
Arch Linux
Linux miniarch 5.0.10-arch1-1-ARCH #1 SMP PREEMPT Sat Apr 27 20:06:45 UTC 2019 x86_64 GNU/Linux

## What went wrong

I ran all the steps to get `eclipselsp` set up but it just silently fails.

## Reproducing the bug

1. Ran all the steps from: https://github.com/eclipse/eclipse.jdt.ls
2. Added this to my config:
```vim
let g:ale_completion_enabled = 1

let g:ale_java_eclipselsp_path = "/home/risto/eclipse.jdt.ls/org.eclipse.jdt.ls.product/target/repository"

let g:ale_linters = {
\   'java': ['eclipselsp'],
\}
```
3. Created a new java project with a HelloWorld class
4. `cd`ed to the project root and opened the class file

```sh
$ java -version
openjdk version "11.0.3" 2019-04-16
OpenJDK Runtime Environment (build 11.0.3+4)
OpenJDK 64-Bit Server VM (build 11.0.3+4, mixed mode)
```

Also tried java 8 but I still get the same issue.

### :ALEInfo

```vim
Current Filetype: java
Available Linters: ['checkstyle', 'eclipselsp', 'javac', 'javalsp', 'pmd']
  Enabled Linters: ['eclipselsp']
 Suggested Fixers: 
  'google_java_format' - Fix Java files with google-java-format.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
  'uncrustify' - Fix C, C++, C#, ObjectiveC, ObjectiveC++, D, Java, Pawn, and VALA files with uncrustify.
 Linter Variables:

let g:ale_java_eclipselsp_executable = 'java'
let g:ale_java_eclipselsp_path = '/home/risto/eclipse.jdt.ls/org.eclipse.jdt.ls.product/target/repository'
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = v:null
let g:ale_completion_delay = 100
let g:ale_completion_enabled = 1
let g:ale_completion_max_suggestions = 50
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 0
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'always'
let g:ale_linter_aliases = {}
let g:ale_linters = {'java': ['eclipselsp']}
let g:ale_linters_explicit = 0
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = v:null
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>>'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History
```

I'm not sure how to debug this or where this is failing. Is the generated executable wrong? Is Ale trying the `eclipselsp` linter at all? Starting the lsp server manually works fine:
```sh
java -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=1044 -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Dlog.level=ALL -noverify -Xmx1G -jar ./plugins/org.eclipse.equinox.launcher_1.5.300.v20190213-1655.jar  -configuration ./config_linux -data ~/eclipse-workspace/foobar/ --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED
```

cc @hsanson

### PR DESCRIPTION
**Pull request by hsanson**
Closes #2468

### CODE REVIEW

File: doc/ale-java.txt [hunk 1]
```diff
@@ -114,8 +114,8 @@ Note: currently, the build can only run when launched with JDK 8. JDK 9 or more
 recent versions can be used to run the server though.
 
 After build completes the files required to run the language server will be
-located inside the folder `org.eclipse.jdt.ls.product/target/repository`.
-Ensure to set |g:ale_java_eclipselsp_path| to the absolute path of that folder.
+located inside the repositoy folder `eclipse.jdt.ls`. Ensure to set
```
Comment (w0rp) [inline doc/ale-java.txt]: Typo here.

```suggestion
located inside the repository folder `eclipse.jdt.ls`. Please ensure you set
```

"Ensure to set" sounds weird, so I suggested "Please ensure you set."

**Comment by w0rp** [comment]
Could we detect the path to the language server configuration automatically? We could probably set it to something like `$HOME . '/eclipse.jdt.ls'` by default.

File: doc/ale-java.txt (hunk 1 above)
Comment (hsanson) [reply doc/ale-java.txt]: Fixed.

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**